### ISSUE TITLE
Properly parallelize the manual ingestion code

### ISSUE THREAD

**Issue by armenzg** [mozilla/treeherder#5807]
I created a management ingestion code for local development that can ingest all tasks for a push and insert them into the DB.

The code I wrote uses asyncio, however, it does not properly paralelize the process. The way it executes it first prepares each individual task for insertion and it then handles all insertions at once. There's the odd DB insertion while the first part happens.

Here's the script and its timing on my machine:
```
real	7m37.738s
user	5m39.210s
sys	0m20.690s
root@ee9ff2ca0679:/app# time ./manage.py ingest push -p mozilla-central -r d6e40de88f3defdc12ef27e64ca73e120b1f10e2 -
```

You can set up the environment by running `docker-compose run backend bash` and you can then execute the command above.

I've asked for some feedback from coworkers and this is what they said:

* None of your tasks are running until you hit the gather on line 93.
* Consider using multiprocessing or concurrent futures for this. Sample example [here](https://github.com/mozilla/treeherder/blob/master/treeherder/perf/management/commands/import_perf_data.py)
  * Notice that multiprocessing is process level not thread level parallelism by default 
* [Project](https://github.com/encode/databases) which can be used for using databases asynchronously
  * I've used the async thread pool executor to handle db calls depending on your queries and DB configuration, the DB may need to serialize all your queries anyway

**Comment by dinolinjob** [comment]
Can I work on this @armenzg ?  If yes, how should i proceed further.

**Comment by armenzg** [comment]
Hello @dinolinjob I mislabeled the issue since I don't know how to solve it, thus, it requires you to research how to write parallelized code.

If you're still interested you can follow these steps:

- Checkt out the code
- Run `docker-compose up`
- On another terminal `docker-compose run backend bash`
- Inside of such container run & watch the output of the following command: `./manage.py ingest push -p mozilla-central -r d6e40de88f3defdc12ef27e64ca73e120b1f10e2 -a`

**Comment by menarulalam** [comment]
Hey @armenzg , I'm going to try to work on this over the weekend.

**Comment by michaelhly** [comment]
Hi @armenzg. Here is my PR for this issue #5915 .

**Comment by armenzg** [comment]
This got fixed on PR #5915 

Unfortunately, there's a lot of warnings for code that will be deprecated soon and it's not actually that fast.

Instead, I want to shift back to ingesting sequentially by using Celerey queues. The ingestion script simply places the requests into the queue and we automatically start up the Celery consumption for pushes and tasks.

Automatic Celery consumption:
https://github.com/mozilla/treeherder/projects/3#card-30391477
Removing the warnings:
https://github.com/mozilla/treeherder/projects/3#card-34313927
